src/main/scala/tile/BaseTile.scala

```diff
@@ -112,6 +112,7 @@ trait HasNonDiplomaticTileParameters {
       //Some(Seq("zicntr")) ++
       Option.when(tileParams.core.useConditionalZero)(Seq("zicond")) ++
       Some(Seq("zicsr", "zifencei", "zihpm")) ++
+      Option.when(tileParams.core.mulDiv.nonEmpty && !tileParams.core.mulDiv.get.divEnabled)(Seq("zmmul")) ++
```

Comment:
Forgot to start a review...

Actually it should be 

```
      Option.when(tileParams.core.mulDiv.nonEmpty)(Seq("zmmul")) ++
```

Check https://github.com/riscv/riscv-isa-manual/issues/869 for more details.

And apply the following diff to the `m` instead

```diff
-val m = if (tileParams.core.mulDiv.nonEmpty) "m" else ""
+val m = if (tileParams.core.mulDiv.nonEmpty  && !tileParams.core.mulDiv.get.divEnabled) "m" else ""
``` 

Also you should update the MISA generation in CSR.scala

Reply:
Hi.
If I understand correctly, there is a diferrence between rv32im_zmmul and rv32im about mul inst when misa.M=0.
If I change ISA of rocket-chip to rv32im_zmmul when mulDiv.nonEmpty, the behaviour of mul inst with misa.M=0 will be changed to not cause illegal instruction exception.
Is it OK?

Reply:
> If I understand correctly, there is a diferrence between rv32im_zmmul and rv32im about mul inst when misa.M=0.

It does exist in that thread, but from my perception it was not the focus of that thread and it was not finally decided. 

From https://github.com/riscv/riscv-isa-manual/issues/869#issuecomment-1186043486, a future `Sm*` extension should come to solve the problem. As it does not exist for now, I assume this detail is undefined, so it is now implementation-defined.

> If I change ISA of rocket-chip to rv32im_zmmul when mulDiv.nonEmpty, the behaviour of mul inst with misa.M=0 will be changed to not cause illegal instruction exception.

I think we should not change the behavior, as it is undefined, and counter-intuitive.

I think we should officially support only two modes: `m` and `zmmul`. As for `m_zmmul`, it is equivalent to `m` and the misa behavior is the same as `m`.

Now I wont insist that we should output `m_zmmul` for `m`. I am OK if you want to just output `m` or `zmmul` in the ISA string. One final commit/comment and I will merge.